File: src/com/leetcode/Medium/FindAllDuplicatesInAnArray.java
package com.leetcode.Medium;

import java.util.*;

// 442. Find All Duplicates in an Array
public class FindAllDuplicatesInAnArray {

    public List<Integer> findDuplicates(int[] nums) {
        List<Integer> list = new ArrayList<Integer>();

        for (int num : nums) {
            int index = Math.abs(num) - 1;
            if (nums[index] < 0)
                list.add(Math.abs(num));

            nums[index] *= -1;
        }

        return list;
    }
}

File: JavaScript/Medium/FindAllDuplicatesInAnArray.js
/**
 * 442. Find All Duplicates in an Array
 * @param {number[]} nums
 * @return {number[]}
 */
const findDuplicates = function (nums) {
  const duplicates = [];

  for (let i = 0; i < nums.length; i++) {
    const index = Math.abs(nums[i]) - 1;

    if (nums[index] < 0) {
      duplicates.push(index + 1);
    } else {
      nums[index] = -nums[index];
    }
  }

  return duplicates;
};